### ISSUE TITLE
Not scrolling elements into view

### ISSUE THREAD

**Issue by DanielGJohnsen** [webdriverio/webdriverio#2164]
I am using webdriver 4.8.0 together with chimp js 0.48.0. And with a new update to chrome, currently using version 59.0.3071.115, apis which should scroll elements into view (i.e. browser.click()) is no longer doing this. browser.scroll() also does nothing, with all combinations of element or long and lat. Is this an issue with a new chrome driver, or webdriverio itself?

**Comment by christian-bromann** [comment]
Please click on the element instead. Usually you don't need to scroll to elements to interact with them. This is definitely not an issue with WebdriverIO. It can not influence that.

**Comment by DanielGJohnsen** [comment]
As i stated, browser.click() does not bring the element into view either, meaning that it can not be interacted with.

**Comment by christian-bromann** [comment]
As I said, you don't need to have the element within the viewport to interact with it. Here an example:

```js
wdio repl chrome                                                                   │s=org.openqa.selenium.logging.LoggingPreferences@3b220df0, browserName=chrome, javascriptEnabled=true, handlesAlerts=t
                                                                                                                       │rue, requestOrigins={name=webdriverio, version=4.8.0, url=http://webdriver.io}}]
[11:54:48]  DEBUG       Queue has stopped!                                                                             │Starting ChromeDriver 2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262) on port 47740
[11:54:48]  DEBUG       You can now go into the browser or use the command line as REPL                                │Only local connections are allowed.
[11:54:48]  DEBUG       (To exit, press ^C again or type .exit)                                                        │11:54:48.582 INFO - Detected dialect: OSS
                                                                                                                       │11:54:48.598 INFO - Done: [new session: Capabilities [{rotatable=true, locationContextEnabled=true, loggingPrefs=org.o
> browser.url("http://webdriver.io")                                                                                   │penqa.selenium.logging.LoggingPreferences@3b220df0, browserName=chrome, javascriptEnabled=true, handlesAlerts=true, re
{ state: 'success',                                                                                                    │questOrigins={name=webdriverio, version=4.8.0, url=http://webdriver.io}}]]
  sessionId: '0bd2c71b-f649-4866-980e-42286c2e9dfd',                                                                   │11:55:48.693 INFO - Executing: [get: http://webdriver.io])
  hCode: 1401475469,                                                                                                   │11:55:53.700 INFO - Done: [get: http://webdriver.io]
  value: null,                                                                                                         │11:56:08.510 INFO - Executing: [find elements: By.cssSelector: footer])
  class: 'org.openqa.selenium.remote.Response',                                                                        │11:56:08.558 INFO - Done: [find elements: By.cssSelector: footer]
  _status: 0 }                                                                                                         │11:56:08.567 INFO - Executing: [tag name: 0 [[ChromeDriver: chrome on MAC (9d23f00d6f65a0477f7ff377fbe52401)] -> css s
> browser.getTagName("footer")                                                                                         │elector: footer]])
'footer'                                                                                                               │11:56:08.580 INFO - Done: [tag name: 0 [[ChromeDriver: chrome on MAC (9d23f00d6f65a0477f7ff377fbe52401)] -> css select
> browser.click("footer")                                                                                              │or: footer]]
{ state: 'success',                                                                                                    │11:56:16.929 INFO - Executing: [find element: By.cssSelector: footer])
  sessionId: '0bd2c71b-f649-4866-980e-42286c2e9dfd',                                                                   │11:56:16.958 INFO - Done: [find element: By.cssSelector: footer]
  hCode: 577853585,                                                                                                    │11:56:16.964 INFO - Executing: [click: 0 [[ChromeDriver: chrome on MAC (9d23f00d6f65a0477f7ff377fbe52401)] -> css sele
  value: null,                                                                                                         │ctor: footer]])
  class: 'org.openqa.selenium.remote.Response',                                                                        │11:56:17.068 INFO - Done: [click: 0 [[ChromeDriver: chrome on MAC (9d23f00d6f65a0477f7ff377fbe52401)] -> css selector:
  _status: 0 }
```

Even the footer was not in the viewport I was able to get information from it. After I clicked on it the browser scrolled it into the viewport.

**Comment by DanielGJohnsen** [comment]
When i console.log() a .click() action i only get this in return

{ sessionId: 'b27fef396b3a8020bdf1c76a8cc6b5db',
  value: null,
  _status: 0 }

Is there something wrong with a configuration?

**Comment by christian-bromann** [comment]
What response do you expect? This looks ok.

**Comment by DanielGJohnsen** [comment]
I am not sure what i expected. But on my end, the problem still persists. I am 100% sure that the element i wish to interact with is in the DOM, but not in the view. All elements that is visible in view are being clicked, but when the element is outside view i get :
unknown error: Element is not clickable at point (998, 914)

I am trying to click a button through browser.click('.next'), and is the only button with that class name.

**Comment by christian-bromann** [comment]
> Element is not clickable at point (998, 914)

This means that another element is laying over the element you want to click. This has nothing to do with the scroll position of the browser if you don't have a skrollr or parallax app.

**Comment by DanielGJohnsen** [comment]
Okay, but how come it can click the element when using firefox, but not chrome? I am not changing any code when i switch browser parameter.

**Comment by mansn** [comment]
I'm also suddenly experiencing issues with this. This has been working before, not sure when it stopped working though.. See below:

```
> browser.url('https://www.ving.se/inspiration');
{ state: 'success',
  sessionId: '43974229-60ac-43c9-acb4-9d7bc388117c',
  hCode: 22746878,
  value: null,
  class: 'org.openqa.selenium.remote.Response',
  _status: 0 }
> $$('div.article-listing.clearfix>a.push')[15].click();
Error: unknown error: Element is not clickable at point (284, 2035)
    at Object.<anonymous> (C:\gitworkspace\UMBAutotest\node_modules\wdio-sync\build\index.js:353:27)
    at evalmachine.<anonymous>:1:47
    at ContextifyScript.Script.runInThisContext (vm.js:25:33)
    at Object.runInThisContext (vm.js:97:38)
    at C:\gitworkspace\UMBAutotest\node_modules\webdriverio\build\lib\commands\debug.js:54:43
    at C:\gitworkspace\UMBAutotest\node_modules\wdio-sync\build\at Object.wait (C:\gitworkspace\UMBAutotest\node_modules\fibers\future.js:449:15)
> $$('div.article-listing.clearfix>a.push')[5].click();
{ state: 'success',
  sessionId: '43974229-60ac-43c9-acb4-9d7bc388117c',
  hCode: 14815901,
  value: null,
  class: 'org.openqa.selenium.remote.Response',
  _status: 0 }
```

Even if I scroll down on the page using browser.scroll(), it still throws the error when trying to click on [15].

selenium server logs:
```
16:28:02.622 INFO - Selenium build info: version: '3.3.1', revision: '5234b32'
16:28:02.622 INFO - Launching a standalone Selenium Server
2017-09-26 16:28:02.645:INFO::main: Logging initialized @221ms to org.seleniumhq.jetty9.util.log.StdErrLog
16:28:02.704 INFO - Driver class not found: com.opera.core.systems.OperaDriver
16:28:02.704 INFO - Driver provider com.opera.core.systems.OperaDriver registration is skipped:
Unable to create new instances on this machine.
16:28:02.704 INFO - Driver class not found: com.opera.core.systems.OperaDriver
16:28:02.704 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
16:28:02.716 INFO - Driver provider org.openqa.selenium.safari.SafariDriver registration is skipped:
 registration capabilities Capabilities [{browserName=safari, version=, platform=MAC}] does not match the current platform WIN10
2017-09-26 16:28:02.764:INFO:osjs.Server:main: jetty-9.2.20.v20161216
2017-09-26 16:28:02.788:INFO:osjsh.ContextHandler:main: Started o.s.j.s.ServletContextHandler@41eaa2{/,null,AVAILABLE}
2017-09-26 16:28:02.894:INFO:osjs.AbstractConnector:main: Started ServerConnector@322d26{HTTP/1.1,[http/1.1]}{0.0.0.0:4444}
2017-09-26 16:28:02.895:INFO:osjs.Server:main: Started @471ms
16:28:02.895 INFO - Selenium Server is up and running
16:28:03.102 INFO - SessionCleaner initialized with insideBrowserTimeout 0 and clientGoneTimeout 1800000 polling every 180000
Selenium started
16:28:05.313 INFO - Executing: [new session: Capabilities [{rotatable=true, locationContextEnabled=true, loggingPrefs=org.openqa.selenium.logging.LoggingPreferences@39b23c, browserName=chrome, javascriptEnabled=true, handlesAlerts=true, requestOrigins={name=webdriverio, version=4.6.2, url=http://webdriver.io}}]])
16:28:05.322 INFO - Creating a new session for Capabilities [{rotatable=true, locationContextEnabled=true, loggingPrefs=org.openqa.selenium.logging.LoggingPreferences@39b23c, browserName=chrome, javascriptEnabled=true, handlesAlerts=true, requestOrigins={name=webdriverio, version=4.6.2, url=http://webdriver.io}}]
Starting ChromeDriver 2.29.461591 (62ebf098771772160f391d75e589dc567915b233) on port 28480
Only local connections are allowed.
16:28:07.608 INFO - Detected dialect: OSS
16:28:07.638 INFO - Done: [new session: Capabilities [{rotatable=true, locationContextEnabled=true, loggingPrefs=org.openqa.selenium.logging.LoggingPreferences@39b23c, browserName=chrome, javascriptEnabled=true, handlesAlerts=true, requestOrigins={name=webdriverio, version=4.6.2, url=http://webdriver.io}}]]
16:28:33.363 INFO - Executing: [get: https://www.ving.se/inspiration])
16:28:35.966 INFO - Done: [get: https://www.ving.se/inspiration]
16:29:04.232 INFO - Executing: [find elements: By.cssSelector: div.article-listing.clearfix>a.push])
16:29:04.279 INFO - Done: [find elements: By.cssSelector: div.article-listing.clearfix>a.push]
16:29:04.314 INFO - Executing: [click: 15 [[ChromeDriver: chrome on XP (46bf23363cde9867718f6d174e5d2e4e)] -> css selector: div.article-listing.clearfix>a.push]])
16:29:04.956 WARN - Exception thrown
org.openqa.selenium.WebDriverException: unknown error: Element is not clickable at point (284, 2035)
  (Session info: chrome=61.0.3163.100)
  (Driver info: chromedriver=2.29.461591 (62ebf098771772160f391d75e589dc567915b233),platform=Windows NT 10.0.14393 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 46 milliseconds
Build info: version: '3.3.1', revision: '5234b32', time: '2017-03-10 09:04:52 -0800'
System info: host: 'NE-LA-SE0190', ip: '10.64.112.213', os.name: 'Windows 10', os.arch: 'x86', os.version: '10.0', java.version: '1.8.0_144'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.29.461591 (62ebf098771772160f391d75e589dc567915b233), userDataDir=C:\Users\9037\AppData\Local\Temp\scoped_dir13232_808}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=61.0.3163.100, platform=XP, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=}]
Session ID: 46bf23363cde9867718f6d174e5d2e4e
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:216)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:168)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:638)
        at org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:274)
        at org.openqa.selenium.remote.RemoteWebElement.click(RemoteWebElement.java:84)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.openqa.selenium.support.events.EventFiringWebDriver$EventFiringWebElement$1.invoke(EventFiringWebDriver.java:332)
        at com.sun.proxy.$Proxy5.click(Unknown Source)
        at org.openqa.selenium.support.events.EventFiringWebDriver$EventFiringWebElement.click(EventFiringWebDriver.java:345)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.openqa.selenium.remote.server.KnownElements$1.invoke(KnownElements.java:64)
        at com.sun.proxy.$Proxy6.click(Unknown Source)
        at org.openqa.selenium.remote.server.handler.ClickElement.call(ClickElement.java:30)
        at org.openqa.selenium.remote.server.handler.ClickElement.call(ClickElement.java:22)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
16:29:04.958 WARN - Exception: unknown error: Element is not clickable at point (284, 2035)
  (Session info: chrome=61.0.3163.100)
  (Driver info: chromedriver=2.29.461591 (62ebf098771772160f391d75e589dc567915b233),platform=Windows NT 10.0.14393 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 46 milliseconds
Build info: version: '3.3.1', revision: '5234b32', time: '2017-03-10 09:04:52 -0800'
System info: host: 'NE-LA-SE0190', ip: '10.64.112.213', os.name: 'Windows 10', os.arch: 'x86', os.version: '10.0', java.version: '1.8.0_144'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.29.461591 (62ebf098771772160f391d75e589dc567915b233), userDataDir=C:\Users\9037\AppData\Local\Temp\scoped_dir13232_808}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=61.0.3163.100, platform=XP, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=}]
Session ID: 46bf23363cde9867718f6d174e5d2e4e
16:29:12.932 INFO - Executing: [find elements: By.cssSelector: div.article-listing.clearfix>a.push])
16:29:12.946 INFO - Done: [find elements: By.cssSelector: div.article-listing.clearfix>a.push]
16:29:12.972 INFO - Executing: [click: 5 [[ChromeDriver: chrome on XP (46bf23363cde9867718f6d174e5d2e4e)] -> css selector: div.article-listing.clearfix>a.push]])
16:29:13.943 INFO - Done: [click: 5 [[ChromeDriver: chrome on XP (46bf23363cde9867718f6d174e5d2e4e)] -> css selector: div.article-listing.clearfix>a.push]]
```

**Comment by mansn** [comment]
My issue is due a bug in chromedriver. I was running chromedrive 2.29, [but it's fixed in 2.32](https://sites.google.com/a/chromium.org/chromedriver/downloads).

I updated through selenium-standalone and it works fine now =).
